src/adapter/src/coord.rs

```diff
@@ -4494,18 +4499,40 @@ impl<S: Append + 'static> Coordinator<S> {
             candidate.join_assign(&ts);
         }
 
-        if when.advance_to_since() {
+        let isolation_level = session.vars().transaction_isolation();
+        let timeline = self.validate_timeline(id_bundle.iter())?;
+        let uses_tables = id_bundle.iter().any(|id| self.catalog.uses_tables(id));
+        let use_timestamp_oracle = (isolation_level == &vars::IsolationLevel::StrictSerializable
+            || uses_tables)
```

Comment:
@mjibson, @frankmcsherry brought up something interesting here. We always use the timestamp oracle for tables so that we can read our own writes. Now that we have separate isolation levels we can remove this requirement. In SERIALIZABLE and below we could just use table `since`s and `upper`s. You may not be able to read your own writes, but they would be faster and not have to wait for table advancements.

Reply:
Yeah, I'd love to get to a spot where we can remove special case logic about tables, or much of it. Ideally we could/would do the same thing for tables as any other source, and just let you opt in to the stronger consistency guarantees (or opt out of them).

Reply:
I'm going to remove the special case for tables then and see if anything breaks.